### ISSUE TITLE
ble_monitor 1.3.1 + pvvx 2.8: mi format and encrypted mi beacon not working?!

### ISSUE THREAD

**Issue by 50494554524F** [custom-components/ble_monitor#312]
i'm using them in homeassistant with xiaomi LYWSD03MMC... 

i used the bindkey stored in the thermometer when it was flashed, i can't get temp values.
then i erased all mi bindkeys and added my bindkey as suggested, same results.

if i use the original firmware and the bindkey i get ble_monitor working perfectly.

can you check, please?!?

thanks

**Comment by Ernst79** [comment]
Did you add it to MiHome? Is it working there? 

If added to MiHome, check the bind key again with Xiaomi cloud token extractor, it might have changed. Adding it to MiHome will trigger a new key generation.

other option is to use the custom advertising format, which can be set in the firmware.

**Comment by 50494554524F** [comment]
never used mihome app, i get the bindkey from pvvx flasher...

i enabled the debug for ble_monitor... i get a tons of "invalid data" rows, these rows disappear when i disable the encryption... so there's a bug somewhere in ble_monitor/firmware/homeassistant, my knowledge finish here and have to put this problem back in the line...

i'll try custom format...

**Comment by Ernst79** [comment]
If you are not planning to use MiHome, it is better to use the custom format anyway. No encryption key required and 'custom' advertising format also sends voltage data of the battery.

**Comment by 50494554524F** [comment]
i have the thermometer working without the encryption, it would be better to use it with the encryption... as i said, using the encryption with original firmware works perfectly... so there's a bug somewhere...

**Comment by Ernst79** [comment]
Ok, I will check it later if I have the same issue.

**Comment by Ernst79** [comment]
Yep, I can confirm your issue. Have to look into it, not sure if it is an issue with the firmware or with our component.

**Comment by Ernst79** [comment]
I found an issue in the format of the data collected with an HCI dump, the first line is a BLE message from stock firmware, the second line from ATC firmware with advertisement type Mi and encryption enabled. As you can see, three bytes are missing. I'll create an issue at the ATC github, my first impression is that is a bug in the firmware. 

```
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
HCI  Evt Len Sub Num Evt  Peer -------MAC-------   Len Len Type Val Len  AD  Xiaomi Frame Product Frame     MAC (LE)        ----------------------PAYLOAD------------  RSSI
type code    evt rep type addr                             flag         type  UUID   ctrl  type    cnt                          cypher            ext.cnt     MAC tag
 A    B  C    D  E    F    G          H            I   J    K   L   M    N     O      P     Q      R           S            
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 04   3E 2A  02  01   00   00  C7 A8 D9 38 C1 A4   1E  02   01  06  1A   16  95 FE  58 58  5B 05   90   C7 A8 D9 38 C1 A4   2D E3 2D 7B 49        00 00 00  EF 49 D6 2E  CE   LYWSD03MMC (default firmware)
 04   3E 27  02  01   00   00  B2 18 8D 38 C1 A4   1B  1A                16  95 FE  58 58  5B 05   2C   B2 18 8D 38 C1 A4   A1 BA 47 2B 4C        00 00 00  1A 7D A5 6F  CD   LYWSD03MMC (ATC, mi-like, encrypted)
```

**Comment by Ernst79** [comment]
Try to update to ATC firmware 2.9. 

I’ll check tomorrow as well.

**Comment by 50494554524F** [comment]
left a message in pvvx pages...

**Comment by Ernst79** [comment]
Note that firmware version 2.9 breaks the parsing in ATC format (custom or ATC1441). Fix is in progress, testing it right now.

**Comment by Ernst79** [comment]
1.4.0 is released with a fix for ATC firmware 2.9. All advertisement formats should work again.

**Comment by Ernst79** [comment]
Closing the issue. Feel free to reopen if you still have issues.
